pkg/kapp/cmd/app/inspect.go

```diff
@@ -91,8 +93,14 @@ func (o *InspectOptions) Run() error {
 			if err != nil {
 				return err
 			}
-
-			resBs, err := historylessRes.AsYAMLBytes()
+			resWithoutManagedFields := historylessRes
+			if !o.ShowManagedFields {
+				resWithoutManagedFields, err = ctldiff.NewResourceWithHistory(historylessRes, nil, nil).ResourceWithoutManagedFields()
```

Comment:
managed fields are not really related to concept of a history within kapp (a set of annotation). we can make a new class that represents resource with/without managed fields.

Reply:
instead of checking on managed fields bool here, we could pass it down into such class so that it owns the behaviour whether to remove or keep managed fields.

Reply:
done